test/test_fake_tensor.py

```diff
@@ -52,22 +52,26 @@ def test_type_as(self):
         self.assertEqual(out.device.type, "cuda")
 
     def test_constructor(self):
-        with enable_torch_dispatch_mode(FakeTensorMode):
+        with enable_torch_dispatch_mode(FakeTensorMode(inner=None)):
```

Comment:
can we just write all of these tests using `FakeTensorMode.push()`? Or maybe @samdow's patch has landed so `with FakeTensorMode()` works now

Reply:
Patch not landed yet (Richard has been hunting down a functorch/dispatch key bug and then I was going to ask him if he had anything to add)--and this actually doesn't work yet because `push` uses `push_torch_dispatch_mode` which causes the error Elias saw late last week (`Creating a new Tensor subclass FakeTensor but the raw Tensor object is already associated to a python object of type Tensor`)

On that note, I forgot to mention Richard debugged that and what's happening is that because push makes every mode have an inner mode (setting BaseTorchDispatchMode if there isn't one set), `detach` has a pyobj associated with it and we end up with this error). The patch fixes this because it removes BaseTorchDispatchMode but it will come up if we nest fake tensor mode with another mode. Long technical way of saying **if we want FakeTensorMode to be composable, we should use push and add `no_dispatch` around all constructors in `torch_dispatch`. I'll try to flag everywhere this will come up**